addons/project/tests/test_project_recurrence.py

```diff
@@ -458,3 +458,62 @@ def test_recurrence_next_dates_year(self):
         self.assertEqual(dates[2], datetime(2023, 11, 30))
         self.assertEqual(dates[3], datetime(2024, 11, 30))
         self.assertEqual(dates[4], datetime(2025, 11, 30))
+
+    def test_recurrence_cron_repeat_after_subtasks(self):
+        parent_task = self.env['project.task'].create({
+            'name': 'Parent Task',
+            'project_id': self.project_recurring.id
+        })
+        domain = [('project_id', '=', self.project_recurring.id)]
+        domain_without_parent = [('project_id', '=', self.project_recurring.id), ('parent_id', '!=', False)]
+        with Form(parent_task.with_context({'tracking_disable': True})) as task_form:
+            with task_form.child_ids.new() as subtask_form:
+                subtask_form.name = 'Test Subtask 1'
+        with freeze_time("2020-01-01"):
+            with Form(parent_task.child_ids.with_context({'tracking_disable': True})) as form:
+                form.description = 'my super recurring task bla bla bla'
+                form.date_deadline = datetime(2020, 2, 1)
+
+                form.recurring_task = True
+                form.repeat_interval = 1
+                form.repeat_unit = 'month'
+                form.repeat_type = 'after'
+                form.repeat_number = 2
+                form.repeat_on_month = 'date'
+                form.repeat_day = '15'
+            subtask = form.save()
+            subtask.planned_hours = 2
+
+            self.assertEqual(subtask.recurrence_id.next_recurrence_date, date(2020, 1, 15))
+            self.assertEqual(self.env['project.task'].search_count(domain), 2)
+            self.assertEqual(self.env['project.task'].search_count(domain_without_parent), 1)
```

Comment:
To avoid to make 2 queries each time you need the number of records according to your domain, we could create a method to fetch all tasks in the project and return the both count that you need 🙂 

```suggestion
            def get_task_counts(project_id)  # XBO don't hesitate to change the name of the method to have a name more explicit for your use case :)
                project_tasks = self.env['project.task'].search([('project_id', '=', project_id)])
                tasks_without_parent_count = len(project_tasks.filtered(lambda task: not task.parent_id))
                return len(project_tasks), tasks_without_parent_count

            project_tasks_count, project_main_tasks_count = get_task_counts(self.project_recurring.id)
            self.assertEqual(project_tasks_count, 2)
            self.assertEqual(project_main_tasks_count, 1)
```

Reply:
I didn't know we should avoid this in tests. Thanks for the remark and code :)